Android/Shared/src/main/java/com/bentley/sample/shared/SampleApplicationBase.kt

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+package com.bentley.sample.shared
+
+import android.annotation.SuppressLint
+import android.app.Application
+import android.content.Context
+import com.github.itwin.mobilesdk.ITMApplication
+
+open class SampleApplicationBase<T: ITMApplication>(private val factory: (context: Context) -> T): Application() {
+    companion object {
+        @SuppressLint("StaticFieldLeak")
+        lateinit var itmApplication: ITMApplication
```

Comment:
Can't this just be type T to get rid of the UNCHECKED_CAST stuff?

Reply:
No, I couldn't figure out a way to make that work in the companion object.

Reply:
Maybe I should get rid of the companion object from this class? So far, the apps aren't using that static instance of ITMApplication, just the get function. The companion object is shared by **all** instances of SampleApplicationBase, regardless of the type parameter, so I now feel that isn't the best design.

Reply:
I guess if it's not used anywhere, then getting rid of it makes sense. I'm somewhat surprised that it's not used anywhere, but the code keeps changing, so I'm not shocked.

Reply:
Well, it was used by the class itself but it can just be a property instead since we will only ever have one instance in an application. The code "keeps changing" because I'm trying to improve it. 😀